src/runtime/HalideRuntime.h

```diff
@@ -1843,6 +1843,13 @@ extern struct halide_profiler_state *halide_profiler_get_state();
  * This function grabs the global profiler state's lock on entry. */
 extern struct halide_profiler_pipeline_stats *halide_profiler_get_pipeline_state(const char *pipeline_name);
 
+/** Collects profiling information. Intended to be called from a timer
+ * interrupt handler if timer based profiling is being used.
+ *  State argument is acquired via halide_profiler_get_pipeline_state.
```

Comment:
What happens if you call this (and other new calls) from thread-based profiling? Even if it's just UB maybe worth calling out.

Reply:
In thread based profiling, Halide fires up a thread to call ```halide_profiler_sample```. It can be called directly as well, though I can't think of a reason to do so. It would simply result in profile samples being collected more often than otherwise I believe.

Reply:
Other calls probably result in undefined references as they are only provided for a timer profiling based runtime.